### DISCUSSION TITLE
Need help with splitting config

### DISCUSSION THREAD

**Discussion by eagerestwolf** [eslint/eslint#17162] [Q&A] [answered]
I am not sure if this is a bug or what; but, for some reason, I cannot import `FlatConfig` objects using relative imports. I have tried regular imports and dynamic imports, but I keep getting `ERR_MODULE_NOT_FOUND` errors. I have included a sample of one of the errors below (with obvious information redacted). The error seems to suggest that ESLint (or Node, not sure which is to blame here) thinks the current working directory is one directory level higher than it is.

```txt
**Oops! Something went wrong! :(

ESLint: 8.39.0

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '[REDACTED]\.eslint\configs\base' imported from [REDACTED]\eslint.config.js
Did you mean to import ../.eslint/configs/base.js?
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:326:11)
    at moduleResolve (node:internal/modules/esm/resolve:945:10)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:525:22)
    at importModuleDynamically (node:internal/modules/esm/translators:110:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:35:14)
```

This is what the `eslint.config.js` looks like (this is an abbreviated version, the actual file is larger)

```js
import baseConfig from "./.eslint/configs/base";
import javascriptConfig from "./.eslint/configs/javascript";
import typescriptConfig from "./.eslint/configs/typescript";
import viteConfig from "./.eslint/configs/vite";

/** @type { import("eslint").Linter.FlatConfig[] } */
export default [
  baseConfig,
  javascriptConfig,
  typescriptConfig,
  viteConfig
];
```

Below is what the TypeScript looks like (just using this one as an example.

```js
import typescript from "@typescript-eslint/eslint-plugin";
import typescriptParser from "@typescript-eslint/parser";

import { IGNORES, TS_FILES, TS_IGNORES, VITE_IGNORES } from "../common/globs";
import typescriptBaseRules from "../rules/typescript/base";
import typescriptNamingConvention from "../rules/typescript/namingConvention";

/** @type {import("eslint").Linter.FlatConfig} */
export default {
  files: [ ...TS_FILES ],
  ignores: [
    ...IGNORES,
    ...TS_IGNORES,
    ...VITE_IGNORES
  ],
  languageOptions: {
    parser: typescriptParser,
    parserOptions: {
      ecmaFeatures: {
        modules: true
      },
      ecmaVersion: "latest",
      project: "./tsconfig.json"
    }
  },
  plugins: [ typescript ],
  rules: {
    ...typescriptBaseRules,
    ...typescriptNamingConvention
  }
}
```

Before anyone asks *"Why would you want to do that?"* I want to do it for a couple of reasons: (1) it makes the `eslint.config.js` file look much neater by abstracting away the endless nesting (not complaining about the nesting, just saying); (2) it makes the config portable because I can simply copy my `.eslint` folder from project to project; and (3) it helps keep the root of the project less cluttered without tons of config stuff.

**Comment by aladdin-add** [comment, marked as answer]
if you are using node.js esm (`"type": "module"` or `.mjs`), the extensions are always required:

```js
import baseConfig from "./.eslint/configs/base.js";
```

**Reply by eagerestwolf** [reply]
I feel like such an idiot. Thank you. I always forget that you have to include the extension when working with pure Node JavaScript code. I'm so used to Typescript and Babel these days.
